rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/profile/AlarmRuleState.java

```diff
@@ -138,10 +139,32 @@ private boolean isActive(long eventTs) {
         switch (alarmRule.getSchedule().getType()) {
             case ANY_TIME:
                 return true;
-            case SPECIFIC_TIME:
-                return isActiveSpecific((SpecificTimeSchedule) alarmRule.getSchedule(), eventTs);
-            case CUSTOM:
-                return isActiveCustom((CustomTimeSchedule) alarmRule.getSchedule(), eventTs);
+            case SPECIFIC_TIME: {
+                SpecificTimeSchedule defaultSchedule = (SpecificTimeSchedule) alarmRule.getSchedule();
```

Comment:
Instead of making these casts, just add a method `getDynamicValue` to the `AlarmSchedule` interface.